backend/src/api/model/event.rs

```diff
@@ -252,6 +252,27 @@ impl AuthorizedEvent {
         ";
         acl::load_for(context, raw_roles_sql, dbargs![&self.key]).await
     }
+
+    /// Returns `true` if the realm has a video block with this video
+    /// OR if the realm has a series block with this event's series.
+    /// Otherwise, `false` is returned.
+    pub(crate) async fn is_referenced_by_realm(&self, path: String, context: &Context) -> ApiResult<bool> {
+        let query = "select exists(\
+            select 1 \
+            from blocks \
+            join realms on blocks.realm = realms.id \
+            where realms.full_path = $1 \
+                and ( \
+                    blocks.video = $2 or \
+                    blocks.series = (select series from events where id = $2) \
+                )\
+            )\
+        ";
+        context.db.query_one(&query, &[&path, &self.key])
```

Comment:
Unfortunately I found a bug still: for the realm path, `/` is passed but in the DB it's stored as `""`. To mimic the behavior of `realmByPath`: just strip trailing slashes. `path.trim_end('/')`.

This causes the bug that clicking on a video on the root page forwards to the direct link.